### DISCUSSION TITLE
Problem with "this" in React component using JSX

### DISCUSSION THREAD

**Discussion by sbusch** [babel/babel#11785] [Q&A]
**Are you aware of any existing issue which tracks the following error?** (or can you help me?)

I get an error when running the build of a `create-react-app` project ("Minified React error #130 - 'this' in JSX arrow function", see <https://github.com/facebook/create-react-app/issues/8738>

```jsx
class ComponentWithJsx extends React.Component {
   First = () => <this.Second />; // <- problem

   Second = () => <div>SUCCESS (ComponentWithJsx)</div>;

   render() {
     return <this.First />;
   }
 }
```

I built a minimal repro by ejecting `create-react-app` and reducing it to the minimal code which triggers the behavior:

  <https://github.com/sbusch/create-react-app-8738-repro.git>

More info in the `README.md` over there and in my comment to this issue here: <https://github.com/facebook/create-react-app/issues/8738#issuecomment-646198606>.

The problem is ultimately caused by the Babel plugin `@babel/plugin-transform-classes`, but it's only happening in conjunction with `@babel/plugin-transform-react-jsx` and `@babel/plugin-proposal-class-properties`.

One can worked around the problem by disabling `@babel/plugin-transform-classes` or by settting the `@babel/plugin-transform-classes` option `{ loose: true }` (or - in a `create-react-app` project - by limiting the `browserslist` for `@babel/preset-env` to browsers which don't need the `@babel/plugin-transform-classes` transform. For the default `browserslist`, this can be achieved by adding `not ie 11`)

Tested with newest versions of `@babel/core` and mentioned packages (see [`package.json`](https://github.com/sbusch/create-react-app-8738-repro/blob/master/package.json))

I couldn't find something related (but it's hard to search through 635 open issues. GitHub's  implementation of search doesn't help here, either).

**Comment by JLHwung** [comment]
@sbusch Thanks for the detailed repro! This is certainly a Babel bug. I have come up with a minimal REPL [example](https://babeljs.io/en/repl#?browsers=defaults&build=&builtIns=false&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4AoctAGxQGc64BBMMOJADxiQDsATRsnQwAdAGFckHr3gBvSnDgAjKHAC8cAEQrNFcoqL8kqjQAoAlOoB8cefsVwiMAK5QecU_YdwAPDAAWwHQiKnAA9FZecOYUigC-FHGUnJCwcHxImCjO1PAsYKRAA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact%2Cenv&prettier=false&targets=&version=7.10.4&externalPlugins=)

Though your example is syntactically correct, as a workaround, you can change `First` to a class method.

```js
class ComponentWithJsx extends React.Component {
   First() { return <this.Second /> }; // <- changed

   Second = () => <div>SUCCESS (ComponentWithJsx)</div>;

   render() {
     return <this.First />;
   }
 }
```

Unfortunately GitHub does not support converting discussion into a bug report. Can you file a bug report? Thanks!

**Reply by sbusch** [reply]
Thank you @JLHwung, I posted an issue here <https://github.com/babel/babel/issues/11806>
